sdk/scheduler/src/main/java/com/mesosphere/sdk/curator/CuratorLocker.java

```diff
@@ -0,0 +1,104 @@
+package com.mesosphere.sdk.curator;
+
+import java.util.concurrent.TimeUnit;
+
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.framework.recipes.locks.InterProcessMutex;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.mesosphere.sdk.scheduler.SchedulerErrorCode;
+import com.mesosphere.sdk.scheduler.SchedulerUtils;
+import com.mesosphere.sdk.specification.ServiceSpec;
+import com.mesosphere.sdk.storage.PersisterUtils;
+
+/**
+ * Gets an exclusive lock on service-specific ZK node to ensure two schedulers aren't running simultaneously for the
+ * same service.
+ */
+public class CuratorLocker {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CuratorUtils.class);
+
+    private static final int LOCK_ATTEMPTS = 3;
+    static final String LOCK_PATH_NAME = "lock";
+
+    private final String serviceName;
+    private final String zookeeperConnection;
+
+    private CuratorFramework curatorClient;
+    private InterProcessMutex curatorMutex;
+
+    public CuratorLocker(ServiceSpec serviceSpec) {
+        this.serviceName = serviceSpec.getName();
+        this.zookeeperConnection = serviceSpec.getZookeeperConnection();
+    }
+
+    /**
+     * Gets an exclusive lock on service-specific ZK node to ensure two schedulers aren't running simultaneously for the
+     * same service.
+     */
+    public void lock() {
+        if (curatorClient != null) {
+            throw new IllegalStateException("Already locked");
+        }
+        curatorClient = CuratorFrameworkFactory.newClient(zookeeperConnection, CuratorUtils.getDefaultRetry());
```

Comment:
From what the unit test has, I think the ACLs are effectively applied to newly written data, so in theory the ACLs written via CuratorPersister shouldn't intersect with the lock.